File: gcpc2025/mexhex/submissions/accepted/yidi.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int n, d;
    cin >> n >> d;

    vector<vector<int>> a(n+1);
    for (int i=0; i<n; i++) {
        int x;
        cin >> x;
        a[x].push_back(i);
    }

    for (int x=0; x<=n; x++) {
        int last = 0;
        bool ok = true;
        for (int i=0; i < a[x].size(); i++) {
            int pos = a[x][i];
            while (i+1 < a[x].size() && a[x][i+1] - a[x][i] < d) i++;

            last = max(last, a[x][i] - d + 1);
            if (last > pos) ok = false;
            last += 2*d;
        }
        if (ok) {
            cout << x << '\n';
            break;
        }
    }
}

File: gcpc2025/mexhex/submissions/accepted/yidi.py
n, d = map(int, input().split())
a = [*map(int, input().split())]
b = [[]for i in range(n+1)]
for i in range(n):
    b[a[i]].append(i)

for x in range(n+1):
    last = 0
    ok = True
    i = 0
    while i < len(b[x]):
        pos = b[x][i]
        while i+1 < len(b[x]) and b[x][i+1] - b[x][i] < d:
            i += 1
        
        last = max(last, b[x][i] - d + 1)
        if last > pos:
            ok = False
        last += 2*d
        i += 1
        
    if ok:
        print(x)
        break
